common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/metrics/MetricProperties.java

```diff
@@ -70,6 +70,10 @@ public Expression getNameExpression() {
 		return spelExpressionParser.parseExpression(nameExpression);
 	}
 
+	public Expression getComputedNameExpression() {
+		return spelExpressionParser.parseExpression(nameExpression != null ? nameExpression : ("'" + name + "'"));
```

Comment:
We usually use `LiteralExpression` for this.
Also, the `nameExpression` can be an `Expression` directly